chirp/drivers/kguv980p.py

```diff
@@ -0,0 +1,2722 @@
+        scan_grp.append(rs)
+
+        for i in range(1, 11):
+            x = str(i)
+            _str = _decode(eval("_settings.scanname"+x))
+            LOG.debug("ScanName %s", i)
+            LOG.debug("is %s", _str)
+            # CPS treats PPPPPP as a blank name as it is the factory reset
+            # value"
+            # The Radio treats PPPPPPP as a blank name and can display 8
+            # chars"
+            # Force Chirp to blank out the scan name if value is PPPPPPP
+            # to match the radio"
+            # Blank out the name if first 6 are spaces or null and the
+            # 7th is a P to handle
+            # firmware peculiarities in handling all 8 characters.
+            if _str[0:7] == "PPPPPPP":
+                _str = ""
+            if self.MODEL != "KG-1000G Plus":
+                if _str[0:7] == "\x00\x00\x00\x00\x00\x00P":
+                    _str = ""
+                elif _str[0:7] == "\x20\x20\x20\x20\x20\x20P":
+                    _str = ""
+            else:
+                if _str[0:6] == "\x00\x00\x00\x00\x00\x00":
+                    _str = ""
+                elif _str[0:6] == "\x20\x20\x20\x20\x20\x20":
+                    _str = ""
+                elif (len(_str) == 2) & (_str[0:2] == "PP"):
+                    _str = ""
+            rs = RadioSetting("scanname" + x, "Scan Name " + x,
+                              RadioSettingValueString(0, 8, _str))
+            scanname_grp.append(rs)
+
+            scngrp = str(i)
+            rs = RadioSetting("scanlower"+scngrp, "Scan Lower "+scngrp,
+                              RadioSettingValueInteger(1, 999,
+                                                       eval(
+                                                        "_settings.\
+                                                        scanlower" +
+                                                        scngrp)))
+            scan_grp.append(rs)
+            rs = RadioSetting("scanupper"+scngrp, "Scan Upper "+scngrp,
+                              RadioSettingValueInteger(1, 999,
+                                                       eval(
+                                                        "_settings.scanupper" +
+                                                        scngrp)))
+            scan_grp.append(rs)
+# remote settings
+        rs = RadioSetting("rc_power", "RC Power",
+                          RadioSettingValueList(
+                           RC_POWER_LIST,
+                           RC_POWER_LIST[_settings.rc_power]))
+        remote_grp.append(rs)
+
+        _code = ''
+        test = int(_settings.ani_edit)
+        _codeobj = '0x{0:0{1}X}'.format(test, 6)
+        LOG.debug("codeobj = %s" % _codeobj)
+        _ani = str(_codeobj)
+        for i in range(2, 8):
+            LOG.debug("ani[i] = %s" % _ani[i])
+            if _ani[i] in dtmfchars:
+                _code += _ani[i]
+        val_ani = int(_code)
```

Comment:
the radio stores the human readable decimal value in hex with stop characters.
so a value of 101 is actually stored as 0x101CCC
There might be an easier way, but the wouxun code does dumb things and since this was working on a previous driver... I didn't spend too much time on it.

Reply:
Okay, but it's stored as an int in the memory on L2467 (_settings.ani_edit), you format it as a hex string on L2465, you stringify it on L2467, then run through the characters, copying them to another string in L2471, and then int() it again on L2472 and then log print it as as a string on L2473. I guess you just want the characters of the string that look like DMTF characters, but this is a really roundabout way to get there.

Also, L2465 adds "0x" in the front, but then you seem to skip the first two characters - why not just not do that?

I know that this works and it probably doesn't seem worth changing it, but if you disappear and someone else has to be able to support this driver (which has happened tons of times in the last fifteen years), it becomes necessary for someone (usually me) to understand what's going on here and the many levels of convolution make that difficult. Some comments would help, so I'd recommend some of that, but fewer rounds of conversions stacked on conversions will go a long way.